transition_sampling/algo/aimless_shooting.py

```diff
@@ -351,26 +416,135 @@ def pick_starting(self, result: ShootingResult) -> np.array:
+            result.fwd["commit"] != result.rev["commit"]
 
-        Uses self.cur_index as the index for this line.
+
+class ResultsLogger:
+    """Class for logging everything from aimless shooting with multiple levels.
+
+    An instance of this class is passed to an AimlessShooting instance in order
+    to log the results to an xyz file and csv file. If constructed with another
+    ResultLogger, that logger is invoked at the same time and duplicates to its
+    respective files.
+
+    This allows one "parent" logger to keep all results in a single pair of
+    files, while many "child loggers" are attached to parallel shootings, each
+    recording only their respective results in a separate pair of files.
+
+    The tracked files are `name`.csv and `name`.xyz. If these files already exist,
+    they are appended to.
+
+    Parameters
+    ----------
+    name
+        The root name of the xyz and csv file
+    base_logger
+        An optional "parent" logger to pass logs up to. If not None, everything
+        logged to this logger is also logged by the `base_logger`
+
+    Attributes
+    ----------
+    name : str
+        The root name of the xyz and csv file to use, e.g. "results"
+    base_logger : ResultsLogger
+        An optional "parent" logger to pass logs up to. If not None, everything
+        logged to this logger is also logged by the `base_logger`
+    cur_index : int
+        The next index that will be written in the CSV.
+    """
+    def __init__(self, name: str, base_logger: ResultsLogger = None):
+        self.name = name
+        self.base_logger = base_logger
+        self._init_csv()
+
+    def log(self, result: ShootingResult, atoms: Sequence[str],
+            frame: np.ndarray, accepted: bool, box_size: Sequence[float]) -> None:
+        """Log results to xyz and csv. Invoke the base logger if we have one.
+
+        This writes all the passed results synchronously to the corresponding
+        XYZ and CSV. The base logger is also invoked, allowing synchronous
+        logging to it as well.
 
         Parameters
         ----------
         result
-            Shooting result of the state to write. Used for fwd and rev basin
-            commits.
+            The shooting result we're logging
+        atoms
+            Periodic table format sequence of atom names
+        frame
+            xyz frame we're logging, each row corresponding to the atoms
+        accepted
+            True if this result was accepted, false otherwise
+        box_size
+            x, y, z of the box size used.
         """
-        columns = [self.cur_index, self.acceptor.is_accepted(result),
-                   result.fwd["commit"], result.rev["commit"],
-                   self.engine.box_size[0], self.engine.box_size[1],
-                   self.engine.box_size[2]]
 
-        with open(self.results_csv, "a") as file:
+        # XYZ
+        comment = f"{result.fwd['commit']}, {result.rev['commit']}"
```

Comment:
I think it would be good to also log the index in the xyz comment just so the connection is concrete and could potentially identify inconsistencies in the future (even if they're unlikely).

Reply:
Fixed